### PR TITLE
183 move hardcoded values to config

### LINKED ISSUE #183: Move hardcoded rule parameters to config
So far we have following hardcoded values in rules:
```
CommonFieldNamesRule
        commonFields = mapOf<String, Property>(
                "id" to StringProperty(),
                "created" to StringProperty("date-time"),
                "modified" to StringProperty("date-time"),
                "type" to StringProperty()
        )

HttpHeadersRule
    PARAMETER_NAMES_WHITELIST = setOf("ETag", "TSV", "TE", "Content-MD5", "DNT", "X-ATT-DeviceId", "X-UIDH",
            "X-Request-ID", "X-Correlation-ID", "WWW-Authenticate", "X-XSS-Protection", "X-Flow-ID", "X-UID",
            "X-Tenant-ID", "X-Device-OS")
            
LimitNumberOfResourcesRule
    PATHS_COUNT_LIMIT = 8
    
LimitNumberOfSubresourcesRule
    SUBRESOURCES_LIMIT = 3
```

We should enable users of Zally to configure these values.
Perhaps after #166 will be merged.

### PR DESCRIPTION
**Pull request by mkulak**
Closes #183 

Also minor refactoring and some bug fixes

### CODE REVIEW

**Comment by fmueller** [comment]
@mkulak Could you resolve the conflicts which your branch has manually please?

**Comment by mkulak** [comment]
@fmueller rebased to latest master.

File: server/src/test/java/de/zalando/zally/rules/HyphenateHttpHeadersRuleTest.kt [hunk 1]
```diff
@@ -9,42 +10,43 @@ import org.junit.Test
 import java.util.HashMap
 
 class HyphenateHttpHeadersRuleTest {
+    val ruleConfig = ConfigFactory.load("rules-config-test.conf")
```
Comment (fmueller) [inline server/src/test/java/de/zalando/zally/rules/HyphenateHttpHeadersRuleTest.kt]: Could we put this into a companion object on package level or sth like this? I miss decent Kotlin language feature understanding here. In each test we have to reference the correct config an instantiate it.

**Comment by fmueller** [comment]
👍

**Comment by fmueller** [comment]
@mkulak Just ping me if you made the changes discussed today.

**Comment by mkulak** [comment]
@fmueller done

**Comment by fmueller** [comment]
👍

**Pull request merged**